gdnative-core/src/nativescript/init/method.rs

```diff
@@ -0,0 +1,491 @@
+//! Method registration
+
+// Temporary for unsafe method registration
+#![allow(deprecated)]
+
+use std::fmt;
+use std::marker::PhantomData;
+
+use crate::core_types::{FromVariant, FromVariantError, Variant};
+use crate::nativescript::class::{NativeClass, RefInstance};
+use crate::object::{Ref, TRef};
+use crate::thread_access::Shared;
+
+use super::ClassBuilder;
+
+/// Builder type used to register a method on a `NativeClass`.
+pub struct MethodBuilder<'a, C, F> {
+    class_builder: &'a super::ClassBuilder<C>,
+    name: &'a str,
+    method: F,
+
+    rpc_mode: RpcMode,
+}
+
+impl<'a, C, F> MethodBuilder<'a, C, F>
+where
+    C: NativeClass,
+    F: Method<C>,
+{
+    pub(super) fn new(class_builder: &'a ClassBuilder<C>, name: &'a str, method: F) -> Self {
```

Comment:
If `class_builder` and `name` don't have the same lifetime, will this still work?

Reply:
Since the lifetime isn't carried on `ClassBuilder` itself, but is inferred by the compiler, the lifetime of `name` simply places a shorter bound on `'a`, which is *then* inferred by the compiler to be the same as `class_builder`. As a result, this does compile: `builder.build_method(&format!("foo{}", 123), method).done()`.